[2022.10.24] feat: projects/:projectId/service/auth post api

ssong-yi opened the pull request:
## 작업 사항
- 팀회의를 거쳐 `service.route`에서 `get`요청 처리 라우터 삭제
- `service.route`에 `post` 요청 처리 라우터 추가
- `project.controller`에서 `secretKey` 인증을 통한 권한 확인 후, 해당 `project`를 응답하는 `showServiceProject` 로직 추가
- mongoose 공식문서에서 권장하는 쿼리 방식에 따른 수정

## 잘 봐주세요
- `project.controller`의 `showServiceProject`를 잘 확인해주세요.

## PR 전 확인사항
* [x] 가장 최신 브랜치를 pull했습니다.
* [x] base 브랜치명을 확인했습니다.
* [x] 코드 컨벤션을 모두 지켰습니다.
* [x] 적절한 라벨이 있습니다.
* [x] assignee가 있습니다.
* [x] (필요하다면) version 업데이트 했습니다.

## 비고
- [mongoose lean()](https://mongoosejs.com/docs/tutorials/lean.html)
- [mongoose exec()](https://mongoosejs.com/docs/promises.html)

Yanguk committed:
feat: signal 서버 구현

client 에서 오는 peer에 대한 시그널을 주고 받게 도와주는 역할을합니다.
socket을 이용하였습니다.

Yanguk committed:
feat: hadleOnLeaveRoom 삭제및 disconnection으로 컨트롤

유저가 퇴장시 소켓 연결이 끊키게하고 disconnect에서 유저 수를 카운트 하였습니다

Yanguk committed:
refactor: 클라인트와의 송수신은 문제없지만 기능추가해야함

인원수 제한 기능 넣기

Yanguk committed:
feat: socket.io를 이용한 실시간 통신 클라이언트랑 연결하기 완료

프론트와 같이 테스트하여 작동이 잘 되었는지 확인하였습니다

Yanguk committed:
refactor: 인원수 제한 기능 삭제

현재 프로젝트 시간이 얼마 없으므로 핵심기능을 위주로 진행하기위하여 부가기능은 삭제하였습니다

Yanguk committed:
refactor: 인원수 제한 로직 삭제

Yanguk committed:
chore: version up

Commit:
feat: projects/:projectId/service/auth post api

- service.route에서 팀회의를 거쳐 get요청 처리 라우터 삭제, post 요청 처리 라우터 추가
- project.controller에서 secretKey 인증을 통한 권한 확인 후 해당 project를 응답하는 showServiceProject 로직 추가
- mongoose 공식문서에서 권장하는 쿼리방식에 따른 수정

Yanguk reviewed (commented):
고생하셨습니다! 덕분에 api가 빠르게 완성되고있어요 👍  코멘트 확인해주세요

ssong-yi committed:
Update src/routes/controllers/project.controller.js

응답에 status code 추가

Co-authored-by: Yanguk <99335782+Yanguk@users.noreply.github.com>

Commit:
refactor: mongoose 권장사항에 따른 수정

get 요청에만 lean()
모든 요청에 exec()

Commit:
Merge branch 'feature/projects-projectId-service-auth-api' of https://github.com/Team-TTT/Circle-BE into feature/projects-projectId-service-auth-api

Commit:
Merge branch 'feature/projects-projectId-service-auth-api' of https://github.com/Team-TTT/Circle-BE into feature/projects-projectId-service-auth-api

Commit:
Merge branch 'feature/socket-signal-server' of https://github.com/Team-TTT/Circle-BE into feature/projects-projectId-service-auth-api

Commit:
refactor: req 수정

- projectId를 req.body가 아니라 req.params에서 가져오는 것으로 수정

Commit:
refactor: router 수정 및 쿼리 수정

- req.params를 사용하기 위해 service.route 파일에서 :projectId를 포함한 경로 설정
- project 정보를 db에서 가져올 때, channels도 함께 가져오도록 수정

rktnsinger reviewed (commented):
송이님 고생 많으셨습니다~! 깔끔하게 작성해주셔서 크게 리뷰드릴게 없었어요 ㅎㅎ 코멘트만 확인 부탁드려요!

ssong-yi reviewed (commented):
형우님 답변 확인 부탁드려요~

Commit:
Merge branch 'develop' of https://github.com/Team-TTT/Circle-BE into feature/projects-projectId-service-auth-api

Commit:
refactor: review 반영 및 버전 업데이트

Yanguk reviewed (approved):
수고하셨습니다~

rktnsinger reviewed (approved):
확인했습니다 고생하셨어요~~!!
